addons/mail/models/mail_thread.py

```diff
@@ -4507,9 +4507,9 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         if strict:
             self._check_can_update_message_content(message.sudo())
 
-        msg_values = {
-            'body': escape(body),  # keep html if already Markup, otherwise escape
-        } if body is not None else {}
+        msg_values = {}
+        # keep html if already Markup, otherwise escape
+        msg_values["body"] = escape(body) + Markup("<span class='o-mail-Message-edited'/>") if body else False
```

Comment:
This is leading to some issues:
- editing empty a message with attachments doesn't show edited even though it should
- body is supposed to be str not False, this is crashing in JS when editing this message again
- adding attachment (no body change) adds "edited", but removing attachment does not